src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -65,7 +65,24 @@ SyntaxNodeAnalysisContext ctx
 				return;
 			}
 
-			var unnamedArgs = GetUnnamedArgs(
+            // Expression trees aren't compatible with named arguments,

```

Comment:
Let's move this into a helper file called `IsExpressionTree` that returns bool and takes your syntax node and the `SemanticModel` as arguments.